Neos.Flow/Classes/Property/TypeConverter/ScalarTypeToObjectConverter.php

```diff
@@ -70,7 +71,7 @@ public function canConvertFrom($source, $targetType)
             return false;
         }
         $methodParameter = array_shift($methodParameters);
-        return $methodParameter['type'] === gettype($source);
+        return TypeHandling::normalizeType($methodParameter['type']) === gettype($source);
```

Comment:
I think, the gettype still should also be normalized (possibly also at other similar places), mainly for the following reason:

http://php.net/manual/en/function.gettype.php
`for historical reasons "double" is returned in case of a float, and not simply "float"`

Reply:
Woot, PHP never stops to surprise me ;)
Anyways, in that case we should probably fix `TypeHandling::normalizeType()` to be in sync with `gettype()`. No need for double conversion IMO

Reply:
And looking at it again, `ReflectionService::getMethodParameters()` should probably already normalize the parameter types so that we don't have to do the above in all cases

Reply:
But then we would again introduce breaky change in behaviour :/ IMO we shouldn't cater to "for historical reasons" crap and rather work around it

Reply:
>And looking at it again, `ReflectionService::getMethodParameters()` should probably already normalize the parameter types so that we don't have to do the above in all cases

This is IMHO a dangerous change and we should think very carefully about it, although I agree in general. Especially we would need to check if let's say the following happens:

```
class OriginalClass {
    __construct(bool $foo)
```

now reflection service returns `boolean` which would probably end up somewhere in proxy building, generating a proxy like:

```
class OriginalClass {
    _construct(boolean $foo)
```

According to @kdambekalns this should still work syntactically. The question is, if this results in an error because the signature doesn't match or if PHP internally normalises this as well and comes to the conclusion that this is in fact the same.

Otherwise we have a problem and should provide a way to get the _raw_ type as hinted.